metricbeat/module/prometheus/remote_write/remote_write.go

```diff
@@ -0,0 +1,127 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package remote_write
+
+import (
+	"io/ioutil"
+	"net/http"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/prompb"
+
+	serverhelper "github.com/elastic/beats/v7/metricbeat/helper/server"
+	httpserver "github.com/elastic/beats/v7/metricbeat/helper/server/http"
+	"github.com/elastic/beats/v7/metricbeat/mb"
+)
+
+func init() {
+	mb.Registry.MustAddMetricSet("prometheus", "remote_write", New)
```

Comment:
If we don't use the `hosts` option to configure the listener we should explicitly disable it to avoid confusions.
```suggestion
	mb.Registry.MustAddMetricSet("prometheus", "remote_write", New,
	  mb.WithHostParser(parse.EmptyHostParser),
	)
```